### DISCUSSION TITLE
Pimcore segments

### DISCUSSION THREAD

**Discussion by triongroup** [pimcore/pimcore#9211] [Q&A / Support] [answered]
During pimcore academy education I've been confused with question "When are segments tracked by Pimcore?".
- When profiling rules with TrackSegment actions are configured.
- Always when opening Pimcore documents tagged with segments.
- Always when opening the detail pages of data objects tagged with segments.
- When SegmentTracker is used by the developer in the custom code.

I've double checked documentation and seen again tutorial videos, but still don't know right answer( Please, help me understand when segments tracked.

**Comment by ckemptner** [comment, marked as answer]
https://pimcore.com/docs/customer-management-framework/current/CustomerSegments.html

https://pimcore.com/docs/customer-management-framework/current/SegmentAssignment.html

**Reply by triongroup** [reply]
I have carefully read documentation and saw again tutorial videos - and exam passed successfully! Thanks!
